vault-plugins/flant_iam_auth/model/authz/authorizator.go

```diff
@@ -65,15 +65,14 @@ func (a *Authorizator) Authorize(authnResult *authn.Result, method *model.AuthMe
 		a.Logger.Debug(fmt.Sprintf("Not found user for %s uuid. Try find service account", uuid))
 		var sa *iam.ServiceAccount
 		sa, err = a.SaRepo.GetByID(uuid)
+		if err == iam.ErrNotFound {
+			return nil, fmt.Errorf("not found iam entity %s", uuid)
+		}
 		if err != nil {
 			return nil, err
 		}
 
-		if sa == nil {
-			return nil, fmt.Errorf("not found iam entity %s", uuid)
-		}
-
-		fullId = user.FullIdentifier
+		fullId = sa.FullIdentifier
```

Comment:
my bad :-) Thanks

Reply:
copy-paste is evil

Reply:
Please, add task for adding test for SA login. No need to add they right now. 

Now we have only users.

Do the same as [here](https://github.com/flant/negentropy/blob/6d13ff5ff5635376052155548e1403cb9517407b/e2e/tests/flow/login_test.go#L84)